src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -189,9 +189,15 @@ static bool HasOrderByHllType(List *sortClauseList, List *targetList);
  * Third, the function pulls up the collect operators in the tree. Fourth, the
  * function finds the extended operator node, and splits this node into master
  * and worker extended operator nodes.
+ *
+ * We also pass plannerRestrictionContext to the optimizer. The context is not
+ * used for any distributed optimization purposes. Instead, it is used to decide
+ * whether the subquery is safe to pushdown. If not, it helps to produce meaningful
+ * error messages for subquery pushdown planning.
```

Comment:
The "is not used" bit seems quite likely to get out of date.  How about 'is primarily used to decide whether ...'?